client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java

```diff
@@ -93,8 +93,24 @@ public void run() {
     }
 
     public void shutdown() {
+        shutdown(0);
+    }
+
+    @Override
+    public void shutdown(long awaitTerminateMillis) {
         this.scheduledExecutorService.shutdown();
         this.consumeExecutor.shutdown();
+        //await to consume
+        if (awaitTerminateMillis > 0) {
+            try {
+                this.consumeExecutor.awaitTermination(awaitTerminateMillis,TimeUnit.MILLISECONDS);
+                if (!this.consumeExecutor.isTerminated()) log.info("There are messages still being consumed in thread pool, but not going to await them anymore. Have awaited for {} ms",awaitTerminateMillis);
```

Comment:
Did you follow our code code guidelines[1] ? Below `if` block is recommended.
```
if {
}
```
http://rocketmq.incubator.apache.org/docs/code-guidelines/